File: frontend/src/types/safetyRule.ts
// Backend: /api/safety-rules
export type SafetyRuleSeverity = 'NORMAL' | 'CAUTION' | 'WARNING' | 'CRITICAL'

export interface SafetyRule {
  id: number
  industryCode: string
  ruleNo: string
  title: string
  content: string
  severity: SafetyRuleSeverity
  sortOrder: number
  active: boolean
}

export interface SafetyRulePayload {
  industryCode: string
  ruleNo: string
  title: string
  content: string
  severity: SafetyRuleSeverity
  sortOrder: number
  active: boolean
}

export interface SafetyRuleListParams {
  industryCode?: string
}

File: backend/src/main/java/com/penocean/ehs/model/SafetyRule.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SafetyRule {

    private Long id;
    private String industryCode;
    private String ruleNo;
    private String title;
    private String content;
    private String severity;            // NORMAL / CAUTION / WARNING / CRITICAL
    private Integer sortOrder;
    private Boolean active;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}
